### PR TITLE
Adds paper implant "cases", cleans up implant case code

### PR DESCRIPTION
**Pull request by BatElite**
[minor][feature][cleanliness]
## About the PR 
This PR adds a paper variant of the implant case for surgically removed implants and those gained from reclaiming bodies. They function the same, except that the paper is qdeled when the implant is taken out. There's also some cleaniness/bugfixing:
-Implant cases now have appropriate implant colours instead of always being blue.
-Implant cases from diggin out /reclaiming implants from people no longer have an inaccessible tracking implant in their contents.

(AFAIK implant cases aren't compatible with glass recyclers, even if they maybe should, so this doesn't result being able to recycle what's meant to be paper.)
afbeelding
afbeelding
(the blood is coloured according to the mob's blood, here a cow, a skeleton & 2 colorful reagent victims. Apparently skeletons don't get white-ish blood?)
## Why's this needed? 
Medbay tends to accumulate a lot glass implant cases if there's frequent cloning/surgery going on and they mostly just clutter up the place. The paper should be easier to click than a bare implant would be, and the deletion thing means there's an opportunity to remove floor clutter if you reuse implants.
Roleplay wise it's more reasonable that surgeons have a supply of or scrap paper around than that they're pulling glass cases out of their asses.

## Changelog

```
(u)BatElite:
(+)Implants cut out of or reclaimed from people now come in disposable paper wads instead of mysterious glass cases. Implant cases should also now be coloured properly based on their contents.
(+)Said wads also have a forensic blood trace of the implant's previous owner, in case you need to be mindful of your traces.
```

### CODE REVIEW

**Comment by BatElite** [comment]
Late idea, but since these visually have blood on them should I try to add a forensic blood trace to the paper wads? I don't know forensics and I have no idea if it'd add anything.

**Comment by pali6** [comment]
> 
> 
> Late idea, but since these visually have blood on them should I try to add a forensic blood trace to the paper wads? I don't know forensics and I have no idea if it'd add anything.

Should be simple I think:
```
implant_case.blood_DNA = the_mob.bioHolder.Uid
implant_case.blood_type = the_mob.bioHolder.bloodType
```

File: code/obj/item/implant.dm [hunk 1]
```diff
@@ -1161,6 +1161,8 @@ THROWING DARTS
 	w_class = 1.0
 	var/implant_type = /obj/item/implant/tracking
 	tooltip_flags = REBUILD_DIST
+	//Whether this is the paper type that goes away when emptied
+	var/disposable = 0
```
Comment (pali6) [inline code/obj/item/implant.dm:1165]:
```suggestion
	var/disposable = FALSE
```

File: code/obj/item/implant.dm [hunk 2]
```diff
@@ -1237,8 +1239,16 @@ THROWING DARTS
 			if (dist <= 1 && src.imp)
 				. += "It appears to contain \a [src.imp.name] with unlimited charges."
 
-/obj/item/implantcase/New()
-	src.imp = new implant_type(src)
+/obj/item/implantcase/New(obj/item/implant/usedimplant = null)
+	if (usedimplant && istype(usedimplant))
+		src.imp = usedimplant
+		imp.set_loc(src)
+		disposable = 1
```
Comment (pali6) [inline code/obj/item/implant.dm:1246]:
```suggestion
		disposable = TRUE
```

**Review by pali6 — APPROVED** [review]

**Comment by BenLubar** [comment]
> (the blood is coloured according to the mob's blood, here a cow, a skeleton & 2 colorful reagent victims. Apparently skeletons don't get white-ish blood?)

Skeletons and cows don't have blood. Instead, they have calcium and milk respectively.

**Comment by BatElite** [comment]
I mean the blood_color on mobs, which is still red for skeletons but white for cows. It's what's used for the decals.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**